### PR TITLE
CATTY-179 Wrong application version in settings

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
Changed version number in `App-Info.plist` and `Settings.bundle/Root.plist` to 0.6.12

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by DhrubojyotiBis1** [comment]
sorry, I didn't follow the commit message style guide if that's a problem then I will open another one.

**Comment by m-herold** [comment]
Thanks for your pull request! Is there a way of reusing any existing variables (e.g. the _MARKETING_VERSION_) in the _plist_ files so that we do not have to change the version number in multiple places when drafting a release?

**Comment by DhrubojyotiBis1** [comment]
Hi @m-herold as you have seen I have changed the PList. I did this because I was unable to find any way to change the app version programmatically. Moreover, MARKETING_VERSION is not a variable it is the default value that I was getting on running the project.

**Comment by m-herold** [comment]
Updating the version number in the settings is definitely a task which needs to be automated. Manually updating this number in multiple files is always a potential source of error (since someone might forget that). However, this can be easily automated since Apple made the _MARKETING_VERSION_ available as an environment variable with Xcode 11.

I'll give you a few suggestions:
Please have a look at the "Add version number to Root.plist" build phase and use the new _$MARKETING_VERSION_ instead of _$CFBundleShortVersionString_. Also it would be a good idea to move this phase above the "Copy Bundle Resources" phase so that it is getting updated immediately (and not after the second build like it currently is). Please also do not edit the _App-Info.plist_.

Thanks!

**Comment by DhrubojyotiBis1** [comment]
Sorry for the late reply and thank you for the advice I didn’t knew this was possible. I will definitely look into the matter to solve the issue.

**Comment by DhrubojyotiBis1** [comment]
@m-herold I have updated the PR based on your suggestions, Root.plist gets updated automatically every time I build. please review.

**Comment by m-herold** [comment]
Thanks, also for updating the order of the build phase. I think now we do not need `CFBundleShortVersionString` anymore, right? Could you please check that. Thanks!

**Comment by DhrubojyotiBis1** [comment]
@m-herold  If you talking about removing `CFBundleShortVersionString`  from the `App-info.plist`  then it's not possible as I think the 'MARKETING_VERSION' is picked from the environment and passed as key to 'CFBundleShortVersionString' which means we are using the 'MARKETING_VERSION' itself through 'CFBundleShortVersionString'

**Comment by m-herold** [comment]
No, I am not talking about the `App-Info.plist` file but the build phase you have updated. I will add a comment directly to the relevant lines of code.

File: src/Catty.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9205,7 +9205,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "CFBundleShortVersionString=`/usr/libexec/PlistBuddy -c \"Print :CFBundleShortVersionString\" \"${SRCROOT}/Catty/App-Info.plist\"`\n/usr/libexec/PlistBuddy \"$SRCROOT/Catty/Settings.bundle/Root.plist\" -c \"set PreferenceSpecifiers:3:DefaultValue $CFBundleShortVersionString\"\n";
+			shellScript = "CFBundleShortVersionString=`/usr/libexec/PlistBuddy -c \"Print :CFBundleShortVersionString\" \"${SRCROOT}/Catty/App-Info.plist\"`\n/usr/libexec/PlistBuddy \"$SRCROOT/Catty/Settings.bundle/Root.plist\" -c \"set PreferenceSpecifiers:3:DefaultValue $MARKETING_VERSION\"\n";
```
Comment (m-herold) [inline src/Catty.xcodeproj/project.pbxproj]: Please remove the unused first line of this build phase:
``CFBundleShortVersionString='/usr/libexec/PlistBuddy -c "Print :CFBundleShortVersionString" "${SRCROOT}/Catty/App-Info.plist"'``

**Comment by DhrubojyotiBis1** [comment]
@m-herold  I have made the changes, yes you were right ` CFBundleShortVersionString` is not required.

**Comment by DhrubojyotiBis1** [comment]
@m-herold seems like the build is failing. I checked the console output of the build it says `Error: junit report not found: './fastlane/test_output/results-BluetoothHelperTests/report.junit'`.

**Comment by m-herold** [comment]
Thanks. Jenkins, retest this please.

**Comment by m-herold** [comment]
Jenkins, retest this please.

**Comment by m-herold** [comment]
Test run is green and everything looks good. Please stick to the project guidelines and squash your commit messages to one commit with a meaningful commit message (i.e. the title of your PR which is perfect). Let me know once you have done this, so that I can directly merge your PR.

**Comment by DhrubojyotiBis1** [comment]
@m-herold I have squashed and renamed. Thanks for your guidance.

**Comment by m-herold** [comment]
Thanks!

**Pull request merged**